iree/compiler/Codegen/Passes.h

```diff
@@ -196,6 +196,8 @@ struct CustomKernelsTargetInfo {
   bool aarch64 = false;
   // Under aarch64: indicates dot-product extension (SDOT, UDOT)
   bool dotprod = false;
+  // Indicates that intrinsics should be used rather than inline asm
```

Comment:
It seems like the thing you link is very similar to what we have here: a manually-generated struct encompassing all the pass options. So I don't follow what you're referring to by a "better way to do this". Agreed that having this be autogenerated would be nice though.

Reply:
This isnt linked to any pass. Basically follow the example where you have the `struct` be a mirror on the pass options, and use that (it might just be a naming thing, but easier to follow that way). `CustomKernelsTargetInfo` makes it look like it is independent of the pass itself.

Reply:
Ah seems like just a naming thing, yeah: it is the pass options for a specific pass. This struct is pre-existing and I believe Benoit is messing with it in another PR, so I'd prefer to break any further modifications out into a separate change.